src/main.cpp

```diff
@@ -2540,16 +2631,35 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 
         if (!tx.IsCoinBase())
         {
-            if (!view.HaveInputs(tx))
+            if (!viewTempCache.HaveInputs(tx)) {
+                // ** In the case of IBD we have to check if the height has changed while we're checking inputs because this thread can continue for a
+                // short while after it has been interrupted and meanwhile the view of the UTXO could have changed after another thread has updated the tip.
+                // ** Additionally and more importantly in the event of a big block DDOS attack, by checking the chain height here, we ensure that
+                //    the thread that is validating the big block will immediately exit and finish up (while still keeping the block on disk in the
+                //    case of a reorg) as soon as the first block makes it through and wins the validation race.
```

Comment:
@dgenr8 it will fail because the outputs will no longer be available because the competing block will have updated the UTXO and spent the coins.  That's assuming the same tx's or many of them are in the same block.  Of course it may not happen 100% of the time so in that event the thread still gets a message from the competing thread to terminate at the bottom of the loop.

Hmmm, just thinking about it more though, an attack block would likely be using txns that were not propagated.  And so any competing blocks wouldn't have those in their block.  So in that case, we'd be relying on the message from the block that finished validation to terminate.  I know it works fine but maybe we should have a backup way of terminating.  It's overkill in my view, but in this part of the code I think a little overkill maybe is warranted.  What do you think?